File: C/exercicios/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float requiredChange;

    do
    {
        requiredChange = get_float("Hom many is your change? ");
        if(requiredChange < 0)
        {
            printf("Type a positive number!\n");
        }
    } while(requiredChange < 0);

    int dollarsToCents = round(requiredChange * 100);
    float change = 0;
    int coinsAmount = 0;

    for(int i = 0; change < dollarsToCents; i++)
    {
        if(change + 25 <= dollarsToCents)
        {
            change += 25;
            coinsAmount++;
        }
        else if(change + 10 <= dollarsToCents)
        {
            change += 10;
            coinsAmount++;
        }
        else if(change + 5 <= dollarsToCents)
        {
            change += 5;
            coinsAmount++;
        }
        else if(change + 1 <= dollarsToCents)
        {
            change += 1;
            coinsAmount++;
        }
    }

    printf("%i\n", coinsAmount);
}

File: Python/pset6/cash.py
from cs50 import get_float

def main():
    amountInCents = get_amount() * 100

    change = 0

    coins = 0

    while change < amountInCents:
        if(change + 25 <= amountInCents):
            change += 25
            coins += 1

        elif(change + 10 <= amountInCents):
            change += 10
            coins += 1

        elif(change + 5 <= amountInCents):
            change += 5
            coins += 1

        elif(change + 1 <= amountInCents):
            change += 1
            coins += 1

    print(coins)

def get_amount():
    while True:
        required_change = get_float("Change owned: ")

        if(required_change < 0):
            print("Type a positive number!\n");
        else:
            break
    return required_change

main()
